src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -411,6 +510,61 @@ private IEnumerable<DocumentTombstone> ReadTombstones(INewDocumentActions action
             }
         }
 
+        internal unsafe LegacyAttachmentDetails ProcessLegacyAttachment(DocumentsOperationContext context, BlittableJsonReaderObject data, ref DocumentItem.AttachmentStream attachment)
+        {
+            BlittableJsonReaderObject bjr;
+            string base64data;
+            string key;
+            if (data.TryGet("Metadata", out bjr) == false ||
+                data.TryGet("Data", out base64data) == false ||
+                data.TryGet("Key", out key) == false)
+            {
+                throw new ArgumentException($"Data of legacy attachment is not valid: {data}");
```

Comment:
Data can be very large, and you need to at lead try to include the key. so we'll know where the error happened.

Reply:
Yes